updater/src/updater.rs

```diff
@@ -95,7 +122,8 @@ pub struct Updater {
 	client: Weak<BlockChainClient>,
 	sync: Weak<SyncProvider>,
 	fetcher: Mutex<Option<fetch::Client>>,
-	operations: Mutex<Option<Operations>>,
+	operations_contract: operations_contract::Operations,
+	do_call: Mutex<Option<Box<Fn(Vec<u8>) -> Result<Vec<u8>, String> + Send + Sync + 'static>>>,
```

Comment:
Could you abstract `do_call`'s type with a type alias? Maybe above `UpdateFilter` you could add:
```pub type DoCallFn = Fn(Vec<u8>) -> Result<Vec<u8>, String> + Send + Sync + 'static>```

Reply:
Sounds good!